app/models/item.rb

```diff
@@ -1,3 +1,13 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  def supermarket_name_by_item
+    self.supermarket.name
+  end
+
+  def total_times_sold
+    self.customers.count
```

Comment:
Great encapsulation of this calculation! I know it's a simple calculation, but by strict MVC  principles this is the place to put it!